src/states/client.rs

```diff
@@ -251,9 +278,12 @@ impl Client {
                          priority: u8)
                          -> Result<(), RoutingError> {
         self.stats.count_user_message(&user_msg);
-        for part in user_msg.to_parts(priority)? {
+        let (hash, parts) = user_msg.to_parts(priority)?;
```

Comment:
Yeah, I did consider both of these.  I thought the way I've done it was slightly cleaner than querying a single part, and having a `get_hash()` function would involve serialising `self` again which seemed wasteful.  But I'm not too bothered between any of these choices - happy to go with popular opinion!